### PR TITLE
Remove dead code in duplicable.rb

### PR DESCRIPTION
**Pull request by casperisfine**
Edit: https://github.com/rails/rails/pull/28544

AFAIK all these feature tests are for pre 2.5.0 MRI, now that Rails requires 2.5.0+, it's no longer useful to test it.

While I was at it I added one exception for `UnboundMethod` that was missing.

Another class that could use it is `Thread`:

```ruby
>> Thread.new{}.dup
TypeError (allocator undefined for Thread)
```

However it would mean requiring the `thread` module, which seems wrong.

cc @rafaelfranca @Edouard-chin

### CODE REVIEW

**Review by Edouard-chin — COMMENTED** [review]
👍

**Comment by kamipo** [comment]
Can you add a test for `UnboundMethod` here?

https://github.com/rails/rails/blob/d2380911847a4d12a55d727b79c94188e5e074ae/activesupport/test/core_ext/object/duplicable_test.rb#L9

File: activesupport/lib/active_support/core_ext/object/duplicable.rb [hunk 1]
```diff
@@ -128,32 +38,12 @@ def duplicable?
...
-    #   Rational(1).duplicable? # => false
-    #   Rational(1).dup         # => TypeError: can't copy Rational
-    def duplicable?
-      false
-    end
+class UnboundMethod
+  # Unbound methods are not duplicable:
+  #
+  #  method(:puts).unbind.duplicable? # => false
+  #  method(:puts).unbinc.dup         # => TypeError: allocator undefined for UnboundMethod
```
Comment (kamipo) [inline activesupport/lib/active_support/core_ext/object/duplicable.rb]:
```suggestion
  #  method(:puts).unbind.dup         # => TypeError: allocator undefined for UnboundMethod
```

**Comment by casperisfine** [comment]
@kamipo done!

**Pull request merged**